Assets/Scripts/Lobby/LobbyManager.cs

```diff
@@ -1,22 +1,49 @@
+using System.Collections;
 using Unity.Services.Lobbies;
+using UnityEngine;
 
 namespace Lobby
 {
-    public class LobbyManager
+    public class LobbyManager : MonoBehaviour
     {
+        private Unity.Services.Lobbies.Models.Lobby CurrentLobby { get; set; }
+        private string lobbyId;
+        
         public Unity.Services.Lobbies.Models.Lobby CreateLobby()
         {
             var task = Lobbies.Instance.CreateLobbyAsync("MyLobby", 4, new CreateLobbyOptions()
             {
                 IsPrivate = true
             });
-            return task.GetAwaiter().GetResult();
+            var lobby = task.GetAwaiter().GetResult();
+            CurrentLobby = lobby;
+            lobbyId = lobby.Id;
+            StartCoroutine(GetLobby(2.5f));
+            return lobby;
         }
 
         public Unity.Services.Lobbies.Models.Lobby JoinLobby(string lobbyCode)
         {
             var task = Lobbies.Instance.JoinLobbyByCodeAsync(lobbyCode);
-            return task.GetAwaiter().GetResult();
+            var lobby = task.GetAwaiter().GetResult();
+            CurrentLobby = lobby;
+            lobbyId = lobby.Id;
+            StartCoroutine(GetLobby(0.1f));
+            return lobby;
+        }
+        
+        private void GetLobby()
+        {
+            var task = Lobbies.Instance.GetLobbyAsync(lobbyId);
+            var lobby = task.GetAwaiter().GetResult();
+            StartCoroutine(GetLobby(0.0001f));
```

Comment:
Whenever we start a coroutine, we actually create a new class. It would be better instead of repeatedly starting new coroutines to use a single coroutine with a while loop inside, which polls repeatedly until our conditions are satisfied. Additionally, we can only poll once per second, so we are currently doing this more often than we need to. 

Finally, WaitForSeconds is actually a class, not a struct; as such, we can definitely benefit from caching our WaitForSeconds before our while loop; the delay should never change because we want to poll at consistent intervals.